contracts/shiden34/lib.rs

```diff
@@ -86,12 +74,25 @@ pub mod shiden34 {
             instance._set_attribute(collection_id.clone(), String::from("name"), name);
             instance._set_attribute(collection_id.clone(), String::from("symbol"), symbol);
             instance._set_attribute(collection_id, String::from("baseUri"), base_uri);
-            instance.payable_mint.max_supply = max_supply;
             instance.payable_mint.price_per_mint = price_per_mint;
             instance.payable_mint.last_token_id = 0;
             instance.payable_mint.max_amount = 1;
+            instance.payable_mint.mint_end = false;
             instance
         }
+
+        #[ink(message)]
+        #[modifiers(only_owner)]
+        pub fn set_code(&mut self, code_hash: [u8; 32]) -> Result<(), PSP34Error> {
```

Comment:
why do you need this function?

Reply:
Our plan is to make the contract upgradable to fix production bugs until the contract is properly audited